tests/e2e/suite_test.go

```diff
@@ -63,13 +56,13 @@ func TestE2E(t *testing.T) {
 	// Run tests through the Ginkgo runner with output to console + JUnit for Jenkins
 	var r []Reporter
 	if framework.TestContext.ReportDir != "" {
+		// Create the directory if it doesn't already exists
+		// NOTE: junit report can be created with new --junit-report flag
+		// https://github.com/kubernetes/kubernetes/blob/4569e646ef161c0262d433aed324fec97a525572/test/e2e_kubeadm/e2e_kubeadm_suite_test.go
```

Comment:
where is this handled now? are we adding this flag anywhere while calling this test suite?

Reply:
Currently we have not yet added this anywhere. 
We need to run the suite using ginkgo to use this flag.

** How this can be done?**
Ginkgo supports generating JUnit reports with
```
ginkgo --junit-report=report.xml
````